flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/TableEnvironment.java

```diff
@@ -280,6 +317,54 @@
 	 */
 	void sqlUpdate(String stmt, QueryConfig config);
 
+	/**
+	 * Gets the current default catalog name of the current session.
+	 *
+	 * @return The current default catalog name that is used for the path resolution.
+	 * @see TableEnvironment#useCatalog(String)
+	 */
+	String getCurrentCatalog();
+
+	/**
+	 * Sets the current catalog to the given value. It also sets the default
+	 * database to the catalog's default one. To assign both catalog and database explicitly
+	 * see {@link TableEnvironment#useDatabase(String, String)}.
+	 *
+	 * <p>This is used during the resolution of object paths. The default path is constructed as
+	 * {@code [current-catalog].[current-database]}. During the resolution, first we try to look for
+	 * {@code [default-path].[object-path]} if no object is found we assume the object path is a fully
+	 * qualified one and we look under {@code [object-path]}.
+	 *
+	 * @param catalogName The name of the catalog to set as the current default catalog.
+	 * @throws CatalogException thrown if a catalog with given name could not be set as the default one
+	 */
+	void useCatalog(String catalogName);
+
+	/**
+	 * Gets the current default database name of the running session.
+	 *
+	 * @return The name of the current database of the current catalog.
+	 * @see TableEnvironment#useDatabase(String, String)
+	 */
+	String getCurrentDatabase();
+
+	/**
+	 * Sets the current default catalog and database. That path will be used as the default one
+	 * when looking for unqualified object names.
+	 *
+	 * <p>This is used during the resolution of object paths. The default path is constructed as
+	 * {@code [current-catalog].[current.database]}. During the resolution, first we try to look for
+	 * {@code [default-path].[object-path]} if no object is found we assume the object path is a fully
+	 * qualified one and we look for {@code [object-path]}.
+	 *
+	 * @param catalogName The name of the catalog to set as the current catalog.
+	 * @param databaseName The name of the database to set as the current database.
+	 * @throws CatalogException thrown if the given catalog and database could not be set as the default ones
+	 */
+	void useDatabase(
```

Comment:
nit: can we add the exception (even though it's runtime exception) to signature to inform users this may throw exception, we've done that for catalog APIs